client/llb/fileop.go

```diff
@@ -683,17 +683,11 @@ func (f *FileOp) Marshal(ctx context.Context, c *Constraints) (digest.Digest, []
 		f.constraints.Platform = p
 	}
 
-	pop, md := MarshalConstraints(c, &f.constraints)
-	pop.Op = &pb.Op_File{
-		File: pfo,
-	}
-
 	state := newMarshalState(ctx)
 	_, err := state.add(f.action, c)
```

Comment:
I wonder if the need for early marshal was that possible modifications in `c` here do not get injected to proto.

Reply:
> I wonder if the need for early marshal was that possible modifications in c here do not get injected to proto.

I didn't see `c` getting modified, but maybe I missed it. Anyway, I changed this to call `addCaps` before the other actions in the loop. Does this look better?